src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs

```diff
@@ -0,0 +1,112 @@
+        }
+
+        private static IOperation Unwrap(IOperation expression)
+        {
+            while (true)
+            {
+                if (expression is IParenthesizedOperation parenthesized)
+                {
+                    expression = parenthesized.Operand;
+                    continue;
+                }
+
+                if (expression is IConversionOperation conversion &&
+                    conversion.IsImplicit)
+                {
+                    expression = conversion.Operand;
+                    continue;
+                }
+
+                return expression;
+            }
+        }
+
+        private static void UnwrapFormatString(
+            IOperation expression, out IOperation unwrapped, out string formatString)
+        {
+            unwrapped = null;
+            formatString = null;
+
+            if (expression is IInvocationOperation { TargetMethod: { Name: nameof(ToString) } } invocation)
+            {
+                if (invocation.Arguments.Length == 1 &&
+                    invocation.Arguments[0].Value.ConstantValue is { HasValue: true, Value: string format })
+                {
+                    unwrapped = invocation.Instance;
+                    formatString = format;
+                    return;
+                }
+
+                if (invocation.Arguments.Length == 0)
+                {
+                    unwrapped = invocation.Instance;
+                    formatString = "";
+                    return;
+                }
+            }
+        }
+
+        private static void UnwrapAlignmentPadding<TExpressionSyntax>(
+            IOperation expression, out IOperation unwrapped, out TExpressionSyntax alignment, out bool negate)
+            where TExpressionSyntax : SyntaxNode
+        {
+            if (expression is IInvocationOperation invocation)
+            {
+                var targetName = invocation.TargetMethod.Name;
+                if (targetName == nameof(string.PadLeft) || targetName == nameof(string.PadRight))
+                {
+                    var argCount = invocation.Arguments.Length;
+                    if (argCount == 1 || argCount == 2)
+                    {
+                        if (argCount == 1 ||
+                            IsSpaceChar(invocation.Arguments[1]))
+                        {
+                            unwrapped = invocation.Instance;
+                            alignment = invocation.Arguments[0].Value.Syntax as TExpressionSyntax;
+                            negate = targetName == nameof(string.PadLeft);
+                            return;
+                        }
+                    }
+                }
+            }
+
+            unwrapped = null;
+            alignment = null;
+            negate = false;
+        }
+
+        private static bool IsSpaceChar(IArgumentOperation argument)
+            => argument.Value.ConstantValue is { HasValue: true, Value: ' ' };
```

Comment:
Due to undefined order of property pattern evaluation, maybe better?

```suggestion
            => argument.Value.ConstantValue.GetValueOrDefault() == ' ';
```

Reply:
Or more idiomatic:

```suggestion
            => argument.Value.ConstantValue == Option.Some(' ');
```

Reply:
there's no issue here.  `Optional<T>` works properly with pattern matching.

Reply:
How do you figure given that the documentation explicitly calls out property pattern matching order as something you should not rely on?

> If it succeeds, then each *property_subpattern* field or property is read and its value matched against its corresponding pattern. [...] The order in which subpatterns are matched is not specified, and a failed match may not match all subpatterns at runtime.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/patterns#property-pattern

Reply:
Oh, Value doesn't throw in this implementation of Optional. That works then :D